app/server/lib/Authorizer.ts

```diff
@@ -165,9 +165,11 @@ export async function addRequestUser(
   if (auth) {
     const tokens = options.gristServer.getAccessTokens();
     const token = await tokens.verify(auth);
+    const userId = token.userId;
+    const user = await dbManager.getUser(userId);
     mreq.accessToken = token;
-    // Once an accessToken is supplied, we don't consider anything else.
-    // User is treated as anonymous apart from having an accessToken.
+    mreq.user = user;
```

Comment:
Is this safe? Could this potentially grant access to material beyond the specific document?

Reply:
I would say that, this is safe as long as the party with access to the provided token is trusted.

Reply:
Ok, I think I understand the concern you raise with that question. It may be possible to access (or update) a another document from a custom widget linked to a document with read (and write) access.

I have the feeling that if this fix "only" reinforce that vulnerability but it does not create it.

I looked at the existing test suite, I remember some related to access limitations but I can't remember if any exist on cross document access limitation/validation.

Reply:
Ok I tested this and I get a `token misuse` from `app/server/lib/Authorizer.ts`.

Reply:
Repository with tests https://github.com/guillett/test-grist-access/blob/main/pages/index.js

(Sorry I did not added a test in grist-core test suite)